crates/composition/src/subgraphs.rs

```diff
@@ -107,6 +110,14 @@ impl Subgraphs {
             subgraphs: self,
         }
     }
+
+    /// Iterates all builtin scalars _that are in use in at least one subgraph_.
+    pub(crate) fn iter_builtin_scalars(&self) -> impl Iterator<Item = StringWalker<'_>> + '_ {
+        ["ID", "String", "Boolean", "Int", "Float"]
+            .into_iter()
+            .filter_map(|name| self.strings.lookup(name))
+            .map(|string| self.walk(string))
+    }
```

Comment:
Are we putting built in scalars into the composition then?  Or is this used for something else.

I'm not even sure what we should be doing around these, just wondering.

Reply:
The reason I had to do this is that we have the `FieldTypeKind` enum. We can either special case the builtin scalars in there with an additional enum variant or add them to the data structure. I don't have strong opinions either way. I also had serialization to protobuf in mind at the time I think, where enums with data aren't that easy to model, so it would have been a DefinitionKind enum and an int. But with serve, we could have a BuiltinScalar variant that contains an enum, easily.

Reply:
One advantage of this is that it's fewer special cases, fewer types, all scalars are scalars (builtin or custom). But there's definitely many cases where we want special treatment for the builtin scalars. So it can make sense to have them separate. I'm a bit undecided.

Reply:
Yeah, we're going to have to special case them _somewhere_ either way.  Not sure where it would be better to do that, but I'm fine with this for now - can always change it if it turns out to not make sense.

Reply:
Sounds good :+1: